dlock/dlock_test.go

```diff
@@ -0,0 +1,48 @@
+package dlock
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/lieut-data/mattermost-plugin-api/dlock/dlocktest"
+	"github.com/stretchr/testify/require"
+)
+
+// TODO(ilgooz): test all branches including related ones to Store errors and ExpireInSeconds.
+// TODO(ilgooz): can move tests from sync/mutex_test.go.
+
+func TestLock(t *testing.T) {
+	dl := New("a", dlocktest.New())
```

Comment:
Maybe it makes sense to also test dlock with a mocked API instead of you in-memory implementation to make sure the API calls work as expected.